src/megaclient.cpp

```diff
@@ -8128,6 +8223,22 @@ void MegaClient::disabletransferresumption(const char *loggedoutid)
 
 void MegaClient::fetchnodes(bool nocache)
 {
+    if (fetchingnodes)
+    {
+        return;
+    }
+
+    WAIT_CLASS::bumpds();
+    fnstats.init();
+    if (sid.size() >= SIDLEN)
+    {
+        fnstats.type = FetchNodesStats::TYPE_ACCOUNT;
+    }
+    else if (publichandle != UNDEF)
```

Comment:
The way we identify if the session is logged into an account or a folder is duplicated many times along the codebase. We should prepare a couple of checker methods like: `bool MegaClient::isLoggedInFolder()` and `isLoggedInAccount()` or similar.